batchlight/src/main/java/co/yello/db/batchlight/androidsupportsqlite/SupportSQLiteBinderConfig.kt

```diff
@@ -0,0 +1,97 @@
+package co.yello.db.batchlight.androidsupportsqlite
+
+import androidx.sqlite.db.SupportSQLiteDatabase
+import co.yello.db.batchlight.Binder
+import co.yello.db.batchlight.BinderConfig
+import co.yello.db.batchlight.sqlAndroidPreparedStatementStartIndex
+import co.yello.db.batchlight.sqlMaxBinds
+
+/**
+ * The [BinderConfig] for Android Support SQLite.
+ *
+ * @property db the Support SQLite database that is being written to.
+ * @property batchStatement the statement that will be executed. For SQLite it will be either REPLACE INTO TABLE_NAME or
+ * INSERT INTO TABLE_NAME
+ *
+ * @constructor
+ *
+ * @param fieldsPerItem the number of fields per row to be inserted.
+ * @param maxFields the maximum number of binds that can happen per statement.
+ */
+class SupportSQLiteBinderConfig(
+    private val db: SupportSQLiteDatabase,
+    private val batchStatement: String,
+    override val fieldsPerItem: Int,
+    override val maxFields: Int
+) : BinderConfig {
+
+    /**
+     * The maximum number of items that can be inserted per statement.
+     */
+    override val maxInsertSize = if (fieldsPerItem > 0) maxFields / fieldsPerItem else 1
+
+    override val maxInsertBinder: Binder by lazy {
+        buildBinder(maxInsertSize)
+    }
+
+    /**
+     * For a SQLite insert statement each item has the format (?, ?, ?)
+     */
+    private val objectStatement = (1..fieldsPerItem).joinToString(
+        prefix = "(",
+        transform = { "?" },
+        separator = ",",
+        postfix= ")"
+    )
```

Comment:
I assumed the Sqlite Delegate was meant to pull this stuff out - was this left by accident or is there a different purpose for SqliteDelegate?

Reply:
Meant to pull that out for now. Was playing with it a bit but it didnt help much. Will push an update with that gone. Thanks